lib/commands/main.ts

```diff
@@ -0,0 +1,69 @@
+import Discord from 'discord.js';
+import commands from '../../data/commands.json';
+import { 
+    ICommand,
+    IExecuteText, 
+    IExecuteCustom, 
+    IExecuteEmbed, 
+    IEmbedField
+} from '../types/command';
+import { getKeyword } from '../helpers';
+import { botRefuses } from '../rng';
+import { isUserAdmin } from '../message';
+
+class Command {
+    public channel: any;
+    public canBeExecuted: boolean;
+    public keyword: string;
+    private isDisabled: boolean;
+    private isModOnly: boolean;
+    private isProtected: boolean;
+    private refusal: string;
+
+    constructor(command:ICommand, msg:Discord.Message) {
+        this.channel = msg.channel;
+        this.isDisabled = command.isDisabled || false;
+        this.isModOnly = command.isModOnly || false;
+        this.isProtected = command.isProtected || false;
+        this.refusal = command.refusal || 'I won\'t execute yout petty command.';
+        this.canBeExecuted = this._canBeExecuted(msg);
+    }
+
+    private _canBeExecuted(msg:Discord.Message) {
+        if (this.isDisabled) {
+            msg.react('🚧');
+            return false;
+        }
+        if (this.isModOnly && !isUserAdmin(msg)) {
+            msg.react('🚫');
+            return false;
+        }
+        if (botRefuses()) {
+            this.channel.send(this.refusal);
+            return false;
+        }
+        return true;
+    }
+}
+
+export class TextCommand extends Command implements IExecuteText {
+    public execute(content:string) {
+        if (this.canBeExecuted) {
+            this.channel.send(content);
```

Comment:
This can be shortened using a short-circuit evaluation: `this.canBeExecuted && this.channel.send(content);`